src/libraries/Common/src/Interop/Unix/System.Globalization.Native/Interop.Normalization.cs

```diff
@@ -12,7 +13,10 @@ internal static partial class Globalization
         [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode, EntryPoint = "GlobalizationNative_IsNormalized")]
         internal static extern int IsNormalized(NormalizationForm normalizationForm, string src, int srcLen);
 
+        internal static int NormalizeString(NormalizationForm normalizationForm, string src, int srcLen, Span<char> dstBuffer) =>
+            NormalizeString(normalizationForm, src, srcLen, ref MemoryMarshal.GetReference(dstBuffer), dstBuffer.Length);
+
         [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode, EntryPoint = "GlobalizationNative_NormalizeString")]
-        internal static extern int NormalizeString(NormalizationForm normalizationForm, string src, int srcLen, [Out] char[] dstBuffer, int dstBufferCapacity);
+        private static extern int NormalizeString(NormalizationForm normalizationForm, string src, int srcLen, ref char dstBuffer, int dstBufferCapacity);
```

Comment:
It is confusing to use `ref char` for buffers; and it is not the fastest option either (requires PInvoke marshalling). You should use `char*` and pin the buffer yourself. 

The PR that you have referenced set a bad precedent that we should not be following.

Reply:
I'll change it to `char*`. @tarekgh In this case, is it okay if I use a Span overload wrapper to hide the pinning from the call site?

Reply:
@jkotas Since we're doing pinning manually anyway, should we pin the string manually as well and change the signature from string to `char*`?

Reply:
Yes, that would be fine with me.